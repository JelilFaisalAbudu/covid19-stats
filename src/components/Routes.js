import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Navbar from './Navbar';
import ConnectedCountryListContainer from '../containers/CountryListContainer';
import CountryDetails from './CountryDetails';
import store from '../redux/store';
import Footer from './Footer';

const Routes = () => (
  <Router>
    <header>
      <Navbar />
    </header>
    <main>
      <Switch>
        <Route exact path="/" component={ConnectedCountryListContainer} />
        <Route
          path="/countries/:countryCountry"
          render={({ match }) => {
            const { dataState } = store.getState();
            return (
              <CountryDetails
                country={
          dataState.data.Countries.find(country => country.Country === match.params.countryCountry)
        }
              />
            );
          }}
        />
      </Switch>
    </main>
    <Footer />
  </Router>
);

export default Routes;
