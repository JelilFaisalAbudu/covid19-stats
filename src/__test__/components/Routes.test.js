import { shallow } from 'enzyme';
import '../setup/setupTests'
import Routes from '../../components/Routes';

const setup = () => shallow(<Routes />);

describe('Routes component', () => {
  let component;

  beforeEach(() => {
    component = setup();
  });

  it('finds the only 1 header element', () => {
    const element = component.find('header');
    expect(element.length).toEqual(1);
  });

  test('renders the nav in the page', () => {
    const item = component.find('.nav');
    expect(item).toMatchSnapshot();
  });

  test('renders the main tag in the page', () => {
    const item = component.find('main');
    expect(item).toMatchSnapshot();
  });

  test('renders the footer in the page', () => {
    const item = component.find('footer');
    expect(item).toMatchSnapshot();
  });
});
