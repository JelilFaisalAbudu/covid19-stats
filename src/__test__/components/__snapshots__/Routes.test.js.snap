// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component renders the footer in the page 1`] = `ShallowWrapper {}`;

exports[`Routes component renders the main tag in the page 1`] = `ShallowWrapper {}`;

exports[`Routes component renders the nav in the page 1`] = `ShallowWrapper {}`;
